clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureBaseClient.groovy

```diff
@@ -46,4 +57,53 @@ public abstract class AzureBaseClient {
       throw new RuntimeException("Failed to create Azure Resource Management Service", e)
     }
   }
+
+  public String getAzureRESTJson(AzureCredentials creds, String baseUrl, String targetUrl, List<String> queryParameters){
+    String result = null
+    String url = AzureUtilities.getAzureRESTUrl(subscriptionId, baseUrl, targetUrl, queryParameters)
+
+    try {
+      // create HTTP Client
+      def config = buildConfiguration(creds)
+      HttpClientBuilder httpClientBuilder = config.create(HttpClientBuilder.class)
```

Comment:
HttpClients are usually created once and re-used.

It seems strange to creating a new one from within this method.

Is there a particular type that should be returned vs. a `String` (I've not looked far enough in the review yet to see where it's being consumed)